### PR TITLE
Update connect-error-using-cluster-nodes-test

### PR DESCRIPTION
**Pull request by bjosv**
Use clusterclient for this testcase instead of clusterclient_async since
the testcase aims to test failures in the initial *blocking* slotmap update.

clusterclient now also handles the flag `--use-cluster-nodes` to update
the slotmap using the command "CLUSTER NODES".

### CODE REVIEW

File: tests/clusterclient.c [hunk 1]
```diff
@@ -54,13 +54,16 @@ void eventCallback(const redisClusterContext *cc, int event, void *privdata) {
...
     int show_events = 0;
     int send_to_all = 0;
 
     int argindex;
     for (argindex = 1; argindex < argc && argv[argindex][0] == '-';
          argindex++) {
-        if (strcmp(argv[argindex], "--events") == 0) {
+        if (strcmp(argv[argindex], "--use-cluster-nodes") == 0) {
+            use_cluster_slots = 0;
+        } else if (strcmp(argv[argindex], "--events") == 0) {
```
Comment (zuiderkwast) [inline tests/clusterclient.c:66]: Nit: If you would add "--use-cluster-nodes" in an else-if (after "--events"), you would get a smaller diff.

File: tests/clusterclient.c [hunk 2]
```diff
@@ -78,14 +81,16 @@ int main(int argc, char **argv) {
...
+        redisClusterSetOptionRouteUseSlots(cc);
+    }
     if (show_events) {
         redisClusterSetEventCallback(cc, eventCallback, NULL);
     }
 
     if (redisClusterConnect2(cc) != REDIS_OK) {
         fprintf(stderr, "Connect error: %s\n", cc->errstr);
-        exit(100);
+        exit(2);
```
Comment (zuiderkwast) [inline tests/clusterclient.c:93]: Maybe we should document this magic number in the comment in the top of the file?

**Review by zuiderkwast — APPROVED** [review]
Good!

File: tests/clusterclient.c (hunk 1 above)
Comment (bjosv) [reply tests/clusterclient.c:66]: I used the same order as in clusterclient_async to have their main() be more alike at start. But they differ in some parts anyway..

File: tests/clusterclient.c (hunk 2 above)
Comment (bjosv) [reply tests/clusterclient.c:93]: Its taken from clientcluster_async, and I guess this is the seconds failure exit() that exits during startup.
The testcase could have been change as an alternative, but I can add some comment here.

File: tests/clusterclient.c [hunk 3]
```diff
@@ -89,8 +89,8 @@ int main(int argc, char **argv) {
     }
 
     if (redisClusterConnect2(cc) != REDIS_OK) {
-        fprintf(stderr, "Connect error: %s\n", cc->errstr);
-        exit(2);
+        printf("Connect error: %s\n", cc->errstr);
+        exit(1);
```
Comment (zuiderkwast) [inline tests/clusterclient.c:94]: 1?

Shouldn't we have the same error codes in the same situations for clusterclient and clusterclient_async?

Btw, It'd still be nice to have the exit codes listed in this comment:

```
/*
 * This program connects to a cluster and then reads commands from stdin, such
 * as "SET foo bar", one per line and prints the results to stdout.
 *
 * The behaviour of the client can be altered by following action commands:
 *
 * !all    - Send each command to all nodes in the cluster.
 *           Will send following commands using the `..ToNode()` API and a
 *           cluster node iterator to send each command to all known nodes.
 *
 */
```

Comment (bjosv) [reply tests/clusterclient.c]: Added comment on top and reverted change.

File: tests/clusterclient.c [hunk 4]
```diff
@@ -8,6 +8,10 @@
  *           Will send following commands using the `..ToNode()` API and a
  *           cluster node iterator to send each command to all known nodes.
  *
+ * Exit statuses this program can return:
+ *   0 - Successful execution of program.
+ *   1 - The required argument "HOST:PORT" is not given.
```
Comment (zuiderkwast) [inline tests/clusterclient.c:13]: Nice list!

Perhaps we should also do `exit(1)` when we print "Unknown argument", i.e. 1 = bad arguments.

Comment (bjosv) [reply tests/clusterclient.c:13]: Yes, that probably good, Ill add that.

**Review by zuiderkwast — APPROVED** [review]

**Pull request merged**